CMakeLists.txt

```diff
@@ -27,16 +28,83 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 
 project(chimera-sdk LANGUAGES C ASM)
 
+# WIESEP: It is important to set the ISA and ABI for the host and the cluster snitch
+set(ABI ilp32)
+set(ISA_CLUSTER_SNITCH rv32im)
+set(ISA_HOST rv32imc)
+
+message(STATUS "[CHIMERA-SDK] ABI                   : ${ABI}")
+message(STATUS "[CHIMERA-SDK] ISA_HOST              : ${ISA_HOST}")
+message(STATUS "[CHIMERA-SDK] ISA_CLUSTER_SNITCH    : ${ISA_CLUSTER_SNITCH}")
+if (${DISASSEMBLE_LIBRARIES})
+  message(STATUS "[CHIMERA-SDK] DISASSEMBLE_LIBRARIES : ON")
+else()
+  message(STATUS "[CHIMERA-SDK] DISASSEMBLE_LIBRARIES : OFF")
+endif()
+
 include(${CMAKE_CURRENT_LIST_DIR}/cmake/Utils.cmake)
 
-add_subdirectory(targets)
-add_subdirectory(hal)
+################################################################################
+# Host Runtime Library                                                         #
+################################################################################
+add_library(runtime_host STATIC)
+
+target_compile_options(runtime_host
+  PRIVATE
+  -O2
+)
+
+target_compile_options(runtime_host
+  PUBLIC
+  -march=${ISA_HOST}
+  -mabi=${ABI}
+)
+
+# WIESEP: Expose common link option
+target_link_options(runtime_host
+  PUBLIC
+  -march=${ISA_HOST}
+  -mabi=${ABI}
+  -nostartfiles
+  -ffreestanding
+)
 
-add_library(chimera-sdk INTERFACE)
-target_link_libraries(chimera-sdk INTERFACE hal)
-target_link_libraries(chimera-sdk INTERFACE runtime)
-target_sources(chimera-sdk INTERFACE $<TARGET_OBJECTS:runtime>)
+################################################################################
+# Snitch Cluster Runtime Library                                               #
+################################################################################
+add_library(runtime_cluster_snitch STATIC)
+
+# WIESEP: Do not export optimization flags
+target_compile_options(runtime_cluster_snitch
+  PRIVATE
+  -O2
+)
+
+target_compile_options(runtime_cluster_snitch
+  PUBLIC
+  -march=${ISA_CLUSTER_SNITCH}
+  -mabi=${ABI}
+)
+
+# WIESEP: Expose common link option
+target_link_options(runtime_cluster_snitch
+  PUBLIC
+  -march=${ISA_CLUSTER_SNITCH}
+  -mabi=${ABI}
+  -nostartfiles
+  -ffreestanding
+)
+
```

Comment:
Same comment as above:
I think we can tie this up one level lower - I'm thinking this might want to live in `targets`.

Reply:
Same, I agree. I will adapt this.